lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1795,9 +1795,9 @@ public void resetStartSerial(int from) {
         }
 
         //fail all pending queued emssages
-        synchronized void fail() {
+        synchronized void fail(ErrorInfo reason) {
             for (QueuedMessage queuedMessage: queue){
-                queuedMessage.listener.onError(new ErrorInfo());
+                queuedMessage.listener.onError(reason);
```

Comment:
Is this method tested anywhere? Is there a way we can add an assertion in a test somewhere that makes sure that the error message is set so we can avoid a future regression?

Reply:
I had added tests for this so I think there must be. I will check the relevant test(s) and add assertions for this particular case. Thanks for bringing it up 👍

Reply:
In retrospect, I think this reason comes from `stateIndication` of `enact` from `State` and not necessarily directly related to our case. I have seen some tests provoke failure call but these aren't the tests I had added. So even if I add a test for this specific case, it won't cover all cases. I think it will probably make sense to add tests for state transitions as general asserting all states will have error codes. But I feel like this is the material for another issue and PR

Reply:
I ended up adding acae4782bc003b6c9a93182facb1d78b7f222ba0 As this is where tests are observing completion handling.

Reply:
Resolving this on Andy's behalf as he approved it now.